sherpa/models/parameter.py

```diff
@@ -41,8 +42,8 @@
 # hugeval = 1.0e+38
 #
 # Use FLT_TINY and FLT_MAX
-tinyval = numpy.float(numpy.finfo(numpy.float32).tiny)
-hugeval = numpy.float(numpy.finfo(numpy.float32).max)
+tinyval = float(numpy.finfo(numpy.float32).tiny)
```

Comment:
I don't think there is a need for explicit conversion here. `numpy.finfo(numpy.float32).tiny` is already a float (a `flaot32` to be precise and numpy will automatically typecast that (keeping the value though!) when used in any of the following calculations. It's not like `numpy.finfo(numpy.float32).tiny` itself will suddenly become larger or smaller through that typecasting.
```suggestion
tinyval = numpy.finfo(numpy.float32).tiny
hugeval = numpy.finfo(numpy.float32).max
```

Reply:
I agree that this looks like unnecessary code here.

Reply:
Pedantry awaits!

Is it guaranteed that Python's `float` type is the same as NumPy's `float32` type? I know that there are "Python will automatically type-convert the values when necessary`, but there are times when it can be useful when moving between Python/C/FORTRAN to be able to say "I really want this type here, not something that will magically contort itself to what we want"

Reply:
actually, I guess the issue is that we are - for the platforms we are interested in - almost guaranteed that `float` and `numpy.float32` have different sizes anyway

```
In [7]: type(numpy.finfo(numpy.float32).eps)
Out[7]: numpy.float32

In [8]: type(float(numpy.finfo(numpy.float32).eps))
Out[8]: float
```

Do we gain or lose anything with conversions here (and my previous comment was actually "the wrong way around" in thinknig about the type of the value storing the data)?

Reply:
Yes, they are different. An explicit conversion does not do any harm, but it's also not necessary. Python and numpy will take care of that automatically if necessary - one of the points of using a high-level dynamically types language as opposed to C (where you would definitely have to know what dtype with variable has) is that you save the developer brainpower needed to think about it. 
And the time used for those automatic conversions is essentially unmeasurably short, so this is purely about saving developer time by not discussing if a conversion is needed or not thinking about it when reading the code; leaving this in or taking it out will have no effect whatsoever on measured runtime or numeric results.